include/shark/ObjectiveFunctions/AbstractObjectiveFunction.h

```diff
@@ -83,9 +87,15 @@ class AbstractObjectiveFunction :  public INameable{
 	typedef PointType SearchPointType;
 	typedef ResultT ResultType;
 
-	typedef SearchPointType FirstOrderDerivative;
+	//if the result type is not an arithmetic type, we assume it is a vector-type->multi objective optimization
+	typedef typename boost::mpl::if_<
+		std::is_arithmetic<ResultT>,
+		SearchPointType,
```

Comment:
If would look more natural to me to replace SearchPointType with RealVector. Well, this is probably not the place for a CompressedRealVector, and a derivative does not make any sense on a discrete search space.